fcd/pass_argrec_remill.cpp

```diff
@@ -403,25 +407,27 @@ bool RemillArgumentRecovery::runOnModule(llvm::Module &module) {
       // convention defines which return register to use.
       LoadReturnRegToRetInsts(cc_func, cc);
 
-      new_funcs.push_back(cc_func);
+      funcs[&func] = cc_func;
     }
   }
 
-  for (auto func : new_funcs) {
+  for (auto func_pair : funcs) {
     // Replace all uses of the old function with the new
     // one with the recovered parameter and return types.
     // Then delete the old function from the module.
-    auto name = TrimPrefix(func->getName());
-    auto old_func = remill::FindFunction(&module, name);
-    UpdateCalls(old_func, func, cc);
-    for (auto &arg : func->args()) {
+    llvm::Function* old_func = func_pair.first;
+    llvm::Function* new_func = func_pair.second;
+    auto is_old_func = [&](llvm::Function* f) { return funcs.find(f) != funcs.end(); };
+    UpdateCalls(old_func, new_func, cc, is_old_func);
+    for (auto &arg : new_func->args()) {
       auto arg_name = TrimPrefix(arg.getName());
-      auto var = remill::FindVarInFunction(func, arg_name);
+      auto var = remill::FindVarInFunction(new_func, arg_name);
       arg.takeName(var);
       removeAttr(arg, llvm::Attribute::Dereferenceable);
     }
-    func->takeName(old_func);
+    assert(old_func->use_empty());
```

Comment:
I would avoid `assert` and use glog's `CHECK` macros for assertion functionality.